dedicated crash

```
No one is playing, shutting down...
Starting a new server instance...
Exception of type `System.NullReferenceException`: Object reference not set to an instance of an object
  at OpenRA.Mods.RA.Server.LobbyCommands.ValidateCommand (OpenRA.Server.Server server, OpenRA.Server.Connection conn, OpenRA.Network.Client client, System.String cmd) [0x00000] in <filename unknown>:0 
  at OpenRA.Mods.RA.Server.LobbyCommands.InterpretCommand (OpenRA.Server.Server server, OpenRA.Server.Connection conn, OpenRA.Network.Client client, System.String cmd) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.InterpretServerOrder (OpenRA.Server.Connection conn, OpenRA.Server.ServerOrder so) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.InterpretServerOrders (OpenRA.Server.Connection conn, System.Byte[] data) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.DispatchOrders (OpenRA.Server.Connection conn, Int32 frame, System.Byte[] data) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadData (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.<Server>m__6B (System.Object _) [0x00000] in <filename unknown>:0 
  at System.Threading.Thread.StartInternal () [0x00000] in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
  at OpenRA.Mods.RA.Server.LobbyCommands.ValidateCommand (OpenRA.Server.Server server, OpenRA.Server.Connection conn, OpenRA.Network.Client client, System.String cmd) [0x00000] in <filename unknown>:0 
  at OpenRA.Mods.RA.Server.LobbyCommands.InterpretCommand (OpenRA.Server.Server server, OpenRA.Server.Connection conn, OpenRA.Network.Client client, System.String cmd) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.InterpretServerOrder (OpenRA.Server.Connection conn, OpenRA.Server.ServerOrder so) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.InterpretServerOrders (OpenRA.Server.Connection conn, System.Byte[] data) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.DispatchOrders (OpenRA.Server.Connection conn, Int32 frame, System.Byte[] data) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadData (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.<Server>m__6B (System.Object _) [0x00000] in <filename unknown>:0 
  at System.Threading.Thread.StartInternal () [0x00000] in <filename unknown>:0 
Platform is Linux
Using Null renderer
Using Null sound engine which disables SFX completely
Available mods:
	cnc: Tiberian Dawn (release-20131223)
	d2k: Dune 2000 (release-20131223)
	ra: Red Alert (release-20131223)
Loading mod: ra
```

```
---------------------------------server.log---------------------------

Initial mod: ra
Initial map: 09341f3ec1523a8e7b0239f9a6a657c16e192784
Client 1: Accepted connection from 94.59.94.247:44328.
Non-blocking send of 4 bytes failed. Falling back to blocking send.
Client 2: Accepted connection from 188.97.72.79:58511.
Non-blocking send of 4 bytes failed. Falling back to blocking send.
Client 3: Accepted connection from 92.37.124.120:57517.
Non-blocking send of 4 bytes failed. Falling back to blocking send.
Client 4: Accepted connection from 174.97.240.117:59005.
Non-blocking send of 4 bytes failed. Falling back to blocking send.
Client 5: Accepted connection from 87.180.113.242:50082.
Client 6: Accepted connection from 99.144.79.102:11639.
Non-blocking send of 4 bytes failed. Falling back to blocking send.
Client 7: Accepted connection from 109.228.85.49:62264.
Dropping client 6 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Player @109.228.85.49:62264 is Ready
Client 8: Accepted connection from 96.50.225.157:50224.
Non-blocking send of 175 bytes failed. Falling back to blocking send.
Client 9: Accepted connection from 90.184.239.124:1063.
Client 10: Accepted connection from 188.106.61.224:57893.
Non-blocking send of 1025 bytes failed. Falling back to blocking send.
Non-blocking send of 2579 bytes failed. Falling back to blocking send.
Non-blocking send of 961 bytes failed. Falling back to blocking send.
Non-blocking send of 2227 bytes failed. Falling back to blocking send.
Client 11: Accepted connection from 176.27.146.137:56727.
Player @90.184.239.124:1063 is Ready
Player @92.37.124.120:57517 is Ready
Player @87.180.113.242:50082 is Ready
Player @92.37.124.120:57517 is NotReady
Player @92.37.124.120:57517 is Ready
Player @92.37.124.120:57517 is NotReady
Player @188.97.72.79:58511 is Ready
Client 12: Accepted connection from 128.70.2.236:61694.
Non-blocking send of 1137 bytes failed. Falling back to blocking send.
Player @96.50.225.157:50224 is Ready
Non-blocking send of 4335 bytes failed. Falling back to blocking send.
Player @109.228.85.49:62264 is Ready
Player @92.37.124.120:57517 is Ready
Player @96.50.225.157:50224 is NotReady
Player @92.37.124.120:57517 is NotReady
Player @109.228.85.49:62264 is NotReady
Player @92.37.124.120:57517 is Ready
Player @109.228.85.49:62264 is Ready
Player @96.50.225.157:50224 is Ready
Client 13: Accepted connection from 79.179.122.26:59277.
Non-blocking send of 1713 bytes failed. Falling back to blocking send.
Dropping client 13 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Client 14: Accepted connection from 37.24.156.137:44335.
Non-blocking send of 1188 bytes failed. Falling back to blocking send.
Player @92.37.124.120:57517 is NotReady
Dropping client 12 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Dropping client 7 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Client 15: Accepted connection from 128.70.2.236:61695.
Player @188.97.72.79:58511 is NotReady
Player @188.97.72.79:58511 is Ready
Non-blocking send of 2146 bytes failed. Falling back to blocking send.
Dropping client 14 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Player @188.97.72.79:58511 is NotReady
Dropping client 15 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Player @188.97.72.79:58511 is Ready
Player @188.97.72.79:58511 is NotReady
Player @188.97.72.79:58511 is Ready
Player @188.97.72.79:58511 is NotReady
Player @188.97.72.79:58511 is Ready
Player @188.97.72.79:58511 is NotReady
Player @188.97.72.79:58511 is Ready
Player @94.59.94.247:44328 is Ready
Player @174.97.240.117:59005 is Ready
Player @94.59.94.247:44328 is NotReady
Dropping client 1 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Dropping client 3 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Dropping client 4 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Dropping client 11 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Dropping client 2 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Dropping client 5 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Dropping client 9 because reading the data failed: System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Receive (System.Byte[] buffer) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadDataInner (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
Initial mod: ra
Initial map: 09341f3ec1523a8e7b0239f9a6a657c16e192784
Client 1: Accepted connection from 83.26.150.157:26222.
Non-blocking send of 4 bytes failed. Falling back to blocking send.
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/1424147-dedicated-crash?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs
+++ b/OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs
@@ -72,7 +72,7 @@ namespace OpenRA.Mods.Common.Server
 
 		public bool InterpretCommand(S server, Connection conn, Session.Client client, string cmd)
 		{
-			if (!ValidateCommand(server, conn, client, cmd))
+			if (server == null || conn == null || client == null || !ValidateCommand(server, conn, client, cmd))
 				return false;
 
 			var dict = new Dictionary<string, Func<string, bool>>

